Pending_updates/soc_004a_human_development_index/soc_004a_human_development_index_processing.py

```diff
@@ -0,0 +1,107 @@
+import requests
+import pandas as pd
+import os
+import sys
+import io
+utils_path = os.path.join(os.path.abspath(os.getenv('PROCESSING_DIR')),'utils')
+if utils_path not in sys.path:
+    sys.path.append(utils_path)
+import util_files
+import util_cloud
+import util_carto
+from zipfile import ZipFile
+import logging
+
+# Set up logging
+# Get the top-level logger object
+logger = logging.getLogger()
+for handler in logger.handlers: logger.removeHandler(handler)
+logger.setLevel(logging.INFO)
+# make it print to the console.
+console = logging.StreamHandler()
+logger.addHandler(console)
+logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+
+
+# name of table on Carto where you want to upload data
+# this should be a table name that is not currently in use
+dataset_name = 'soc_004a_human_development_index' #check
+
+logger.info('Executing script for dataset: ' + dataset_name)
+# create a new sub-directory within your specified dir called 'data'
+# within this directory, create files to store raw and processed data
+data_dir = util_files.prep_dirs(dataset_name)
+   
+'''
+Download data and save to your data directory
+'''
+logger.info('Downloading raw data')
+# insert the url used to download the data from the source website
+url = 'http://hdr.undp.org/sites/default/files/hdro_statistical_data_tables_1_15_d1_d5.xlsx' #check
+
+# read in data to pandas dataframe
+r = requests.get(url)
+df = pd.read_excel(io.BytesIO(r.content), encoding='utf8')
+
+# save unprocessed source data
+raw_data_file = os.path.join(data_dir, os.path.basename(url))
+df.to_excel(raw_data_file, header=False, index=False)
+
+'''
+Process data
+'''
+# store the information in the metadata rows 
+titles = list(df.iloc[3].fillna(''))
+units = list(df.iloc[4].fillna(''))
+years = list(df.iloc[5].fillna(''))
+headers = []
+for title, year, unit in zip(titles, years, units):
+    headers.append((year + ' ' + title + ' ' + unit).strip())
+
+# drop metadata rows
+df = df[7:]
+# assign correct headers to the columns 
+df.columns = headers
+# remove columns with unimportant information or no values at all 
```

Comment:
I don't want to call this information 'unimportant'
```suggestion
# remove unnamed columns because they contain flags to reference footers that are no longer in the table, or contain no values at all
```